components/brave_search/browser/brave_search_host.cc

```diff
@@ -7,24 +7,91 @@
 
 #include <utility>
 
+#include "base/strings/stringprintf.h"
+#include "net/base/load_flags.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+namespace {
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("brave_search_host", R"(
+      semantics {
+        sender: "Brave Search Host Controller"
+        description:
+          "This controller is used as a backup search "
+          "provider for users that have opted into this feature."
+        trigger:
+          "Triggered by Brave search if a user has opted in."
+        data:
+          "Local backup provider results."
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        setting:
+          "You can enable or disable this feature on chrome://flags."
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+const unsigned int kRetriesCountOnNetworkChange = 1;
+static GURL backup_provider_for_test;
+}  // namespace
+
 namespace brave_search {
 
-BraveSearchHost::BraveSearchHost() : weak_factory_(this) {}
+void BraveSearchHost::SetBackupProviderForTest(const GURL& backup_provider) {
+  backup_provider_for_test = backup_provider;
+}
+
+BraveSearchHost::BraveSearchHost(
+    scoped_refptr<network::SharedURLLoaderFactory> factory)
+    : shared_url_loader_factory_(std::move(factory)), weak_factory_(this) {}
 
 BraveSearchHost::~BraveSearchHost() {}
 
-void BraveSearchHost::FetchBackupResults(const std::string& query_string,
+void BraveSearchHost::FetchBackupResults(const std::string& query,
                                          const std::string& lang,
                                          const std::string& country,
                                          const std::string& geo,
                                          FetchBackupResultsCallback callback) {
-  // TODO(sergz): make a call to fetch a response and remove that callback call
-  OnFetchBackupResults(std::move(callback), "response");
+  auto request = std::make_unique<network::ResourceRequest>();
+
+  if (backup_provider_for_test.is_empty()) {
+    std::string spec(
+        base::StringPrintf("https://www.google.com/search?q=%s&hl=%s&c=%s",
```

Comment:
Should `c=` be `gl=`? I think that's how Google expects the country parameter.

Reply:
Thanks, will be fixed here https://github.com/brave/brave-browser/issues/15130